src/components/common/Navbar.tsx

```diff
@@ -0,0 +1,108 @@
+import Link from 'next/link';
+import styles from './Navbar.module.css';
+import LogoIcon from '#/icons/icon-logo.svg';
+import ProfileIcon from '#/icons/icon-profile.svg';
+import AlarmIcon from '#/icons/icon-alarm.svg';
+import RedEllipse from '#/icons/icon-redEllipse.svg';
+import { useState } from 'react';
+import AlarmModal from '@/components/Navbar/AlarmModal';
+import ProfileDropDown from '@/components/Navbar/ProfileDropDown';
+
+const USER_DATA = { name: '종민박' };
+const alarmData = {
+  notifications: [
+    {
+      id: 0,
+      content: '함께하면 즐거운 댄스(2023-01-14 15:00:~18:00) 예약이 거절되었어요.',
+      status: 'declined',
+      createdAt: '2024-01-26T06:23:36.209Z',
+    },
+    {
+      id: 1,
+      content: '함께하면 슬픈 댄스(2023-01-15 15:00:~18:00) 예약이 승인되었어요.',
+      status: 'approve',
+      createdAt: '2024-01-26T06:23:36.209Z',
+    },
+    {
+      id: 2,
+      content: '함께하는 종민 댄스(2023-01-16 15:00:~18:00) 예약이 거절되었어요.',
+      status: 'declined',
+      createdAt: '2024-01-26T06:23:36.209Z',
+    },
+  ],
+  totalCount: 3,
+};
+
+function Navbar() {
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleDropDownClick = () => {
+    setIsOpen((prev) => !prev);
+    if (isOpen) {
+      setTimeout(() => {
+        setIsDropDownOpen((prev) => !prev);
+      }, 250);
+    } else {
+      setIsDropDownOpen((prev) => !prev);
+    }
+  };
+
+  const handleBlurDropDown = () => {
+    if (isOpen) {
+      setIsOpen(false);
+    }
+    setTimeout(() => {
+      setIsDropDownOpen(false);
+    }, 250);
+  };
+
```

Comment:
굳이 state를 두 개쓰고 함수를 두 개 쓸 필요가 있을까요?

```suggestion
  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
  const [isModalOpen, setIsModalOpen] = useState(false);

  const handleDropDownClick = () => {
    if (isDropDownOpen) {
      setTimeout(() => {
        setIsDropDownOpen((prev) => !prev);
      }, 250);
    } else {
      setIsDropDownOpen((prev) => !prev);
    }
  };

```

이것도 똑같이 동작하는 거 같아서 함수 이름만 handleDropDownToggle 이런 식으로 바꾸고 onblur 함수에도 걸어주면 될 거 같습니다.

Reply:
다시 확인해봤는데 blur에 대한 함수와 click에 대한 함수를 나누는 건 좋은 거 같습니다. 

```ts
  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
  
  const handleDropDownClick = () => {
    setIsDropDownOpen((prev) => !prev);
  };

  const handleDropDownBlur = () => {
    setTimeout(() => {
      setIsDropDownOpen(false);
    }, 250);
  };

```
이게 더 적합할 거 같네요.

Reply:
이게 아마 애니메이션 넣으려고 isOpen이라는 state 값을 하나 더 만들어서 props로 먼저 내려줘서 style을 바꾸게 하고 
그다음에 isDropDownOpen state값을 바꿔서 애니메이션이 다 끝난 다음 드롭다운을 끄게 하려고 만들어준 건데
유담님껏 처럼 하면 드롭다운 사라지는 애니메이션 없이 바로 사라지는 오류가 있습니다.

Reply:
어렵네여